openedx_tagging/core/tagging/models/system_defined.py

```diff
@@ -0,0 +1,271 @@
+""" Tagging app system-defined taxonomies data models """
+import logging
+from typing import Any, List, Type, Union
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.db import models
+
+from openedx_tagging.core.tagging.models.base import ObjectTag
+
+from .base import Tag, Taxonomy, ObjectTag
+
+log = logging.getLogger(__name__)
+
+
+class SystemDefinedTaxonomy(Taxonomy):
+    """
+    Simple subclass of Taxonomy which requires the system_defined flag to be set.
+    """
+
+    class Meta:
+        proxy = True
+
+    @property
+    def system_defined(self) -> bool:
+        """
+        Indicates that tags and metadata for this taxonomy are maintained by the system;
+        taxonomy admins will not be permitted to modify them.
+        """
+        return True
+
+
+class ModelObjectTag(ObjectTag):
```

Comment:
How does this interact with [`ContentObjectTag`](https://github.com/open-craft/edx-platform/blob/76d8dcca420d45000ea9ae9326c3ab9a6de30f95/openedx/features/content_tagging/models.py#L85-L104)?

I don't really like that sometimes the type of taxonomy (system-defined or not) determines which subclass of `ObjectTag` to use (`ModelObjectTag`), and other times the type of content (XBlock or Course) determines it (`ContentObjectTag`). I know that they're both driven by the taxonomy actually, since ContentTaxnonomy is a type of Taxonomy too, but it seems like there's some conceptual confusion. (I'm not asking for changes just yet, but trying to understand what the idea is here).

Reply:
Hmm.. maybe ModelObjectTag's functionality should be in a mixin instead, so that we can have ModelContentObjectTags..

But that probably also means that the SystemTaxonomy / ModelSystemTaxonomy functionality should be in a mixin too, same reason.. is this getting to complicated?

Reply:
Cleanup task: https://github.com/openedx/modular-learning/issues/85